lib/render/RayCaster.cpp

```diff
@@ -1398,15 +1414,13 @@ int RayCaster::_updateVertCoordsTexture( const glm::mat4& MV )
         return MEMERROR;
     }
 
-    const float* vc = _userCoordinates.vertCoords;
     glm::vec4 posModel( 1.0f );
+    float*    posModelPtr = glm::value_ptr( posModel );
```

Comment:
Again, I used a pointer here to enable `memcpy()` to avoid three individual copies. Do you think it'd address your concern if I use a local scope to limit the lifespan of this pointer?

Reply:
A couple things:

This `memcpy` is probably optimized out by the compiler to three individual copies as a function call is much more expensive. A function call by itself requires copying the current instruction pointer, stack pointer, the arguments, and all saved registers to the stack. Then you have the overhead of moving the instruction pointer far away from the current cache. And then you have all the overhead of the memcpy algorithm.

You can call `memcpy(glm::value_ptr( posModel ), ...)`

You are essentially rendering to a texture on the CPU. This could be done on the GPU.

GLM is "layout compatible" with c arrays so you could cast float[x][3] to vec3[] and vice versa instead of copying data.